NOTICE.txt

```diff
@@ -1,7 +1,16 @@
-OpenSearch
-Copyright 2021 OpenSearch Contributors
-This product includes software, including Kibana source code, developed by Elasticsearch (http://www.elastic.co).
-Copyright 2012-2021 Elasticsearch B.V.
```

Comment:
Was this in the original Kibana license? I think we should then reproduce it as is.

Reply:
Part of this was, yes. Do we need to include their NOTICE header? https://github.com/elastic/kibana/blob/v7.10.2/NOTICE.txt

Reply:
I am not 100% sure, let's ask @hyandell?

Reply:
We need to include the relevant NOTICE text for any Apache-2.0 with a NOTICE content that we bring in. Make sure it's the NOTICE text at that time; i.e. don't include that project's NOTICE 2 years later on just because it's the latest on their main branch.

The only proviso is if we have removed the content it refers to. 

In the case of this NOTICE file PR in general, my concerns are:

* Removing our copyright year:  Not hugely concerned. We don't usually put copyright years on our OSS, but we included it in this case to avoid it seeming odd when the original codebase did use years.
* Changing the top "This product includes" language. I'm not seeing a great need to do this.
* Changing Elastic's copyright date. This is concerning and should only be done if you're identifying that nothing post 2018 was included in the repository. 
* Adding the line about Joda. Is this true for this project, or is this being copied from another project?
* Removing other items in the NOTICE. This should only be done if you've conducted a review to confirm they're no longer there. I think we should do that one GitHub issue a time, with a PR for each, so that we are properly tracking the review we've done to confirm that that notice-element no longer, or never did, apply.

Reply:
We cleared the whole removing year for _our_ copyright and the language earlier.

Reply:
Thanks Henri for the clarifications. I'm deferring the NOTICE changes for now since they're not required and I've opened #765 to address discrepancies that exist between the current NOTICE file and the generator script.